@gwwatkin You might want to look at those last 3 errors from flake8
```
lsqecc\patches\ancilla_region_routing.py:32:5: E741 ambiguous variable name 'l'
lsqecc\patches\lattice_surgery_computation_composer.py:170:33: F841 local variable 'blank_slice' is assigned to but never used
lsqecc\pipeline\lattice_surgery_compilation_pipeline.py:14:1: F822 undefined name 'VisualArrayCell' in __all__
```

--- a/README.md
+++ b/README.md
@@ -32,6 +32,21 @@ This project is under very active development. We hope to see in the very near f
 ## Contributing
 To get started, we recommend cloning the repo and trying out the compiler. The `debug` folder has a lot of examples of things you can do inside this project. To see where we are at, in terms of development, check out our [project board](https://github.com/orgs/latticesurgery-com/projects/1).
 
+### Dev setup
+In the project's root folder, create pip virtual environment with: 
+
+`$ python -m venv ./venv`
+
+Activate the venv:
+
+`$ source ./venv/bin/activate`
+
+Then install the package in edit mode:
+
+`$ pip install -e .`
+
+Now you can use this venv to work the package, from the command line and in an IDE. Make sure to select the venv you just created as the python environment for the project in your IDE.
+
 ## References
 <a id="1">[1]</a> 
 Alexandru Paler and Austin G Fowler. 

--- a/lsqecc/patches/ancilla_region_routing.py
+++ b/lsqecc/patches/ancilla_region_routing.py
@@ -29,21 +29,24 @@ def get_pauli_op_listing(
         return patch
 
     # TODO check overlapping with representative and document
-    l = list(
+    operators_of_cells_in_patch = list(
         filter(
             lambda cell: cell in patch_pauli_operator_map, get_patch_of_cell_from_patch(cell).cells
         )
     )
 
-    if len(l) == 0:
+    if len(operators_of_cells_in_patch) == 0:
         return None
 
     r = lattice.getPatchRepresentative(cell)
-    if l[0] != r:
+    if operators_of_cells_in_patch[0] != r:
         raise Exception(
-            "Non patch repr cell associated with operator: " + str(l[0]) + ". Repr is " + str(r)
+            "Non patch repr cell associated with operator: "
+            + str(operators_of_cells_in_patch[0])
+            + ". Repr is "
+            + str(r)
         )
-    return l[0] if len(l) > 0 else None
+    return operators_of_cells_in_patch[0] if len(operators_of_cells_in_patch) > 0 else None
 
 
 def make_graph_of_free_cells(lattice: patches.Lattice) -> igraph.Graph:

--- a/lsqecc/patches/lattice_surgery_computation_composer.py
+++ b/lsqecc/patches/lattice_surgery_computation_composer.py
@@ -1,7 +1,7 @@
 import copy
 import enum
 import uuid
-from typing import Dict, List, Optional, Tuple
+from typing import Dict, List, Optional, Tuple, cast
 
 import lsqecc.logical_lattice_ops.logical_lattice_ops as llops
 import lsqecc.patches.patches as patches
@@ -168,7 +168,7 @@ class LatticeSurgeryComputation:
         sim = lps.PatchSimulator(logical_computation)
 
         with comp.timestep() as blank_slice:
-            pass
+            cast(object, blank_slice)  # no-op
 
         for logical_op in comp.logical_computation.ops:
             if logical_op.does_evaluate():

--- a/lsqecc/pipeline/lattice_surgery_compilation_pipeline.py
+++ b/lsqecc/pipeline/lattice_surgery_compilation_pipeline.py
@@ -11,7 +11,7 @@ from lsqecc.lattice_array import sparse_lattice_to_array
 
 GUISlice = List[List[Optional[vac.VisualArrayCell]]]  # 2D array of cells
 
-__all__ = ["compile_file", "VisualArrayCell", "GUISlice"]
+__all__ = ["compile_file", "GUISlice"]
 
 
 def compile_file(

